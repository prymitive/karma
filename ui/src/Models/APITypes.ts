export type AlertStateT = "unprocessed" | "active" | "suppressed";

export type LabelsT = { [key: string]: string };

export interface AlertmanagerSilenceMatcherT {
  name: string;
  value: string;
  isRegex: boolean;
  isEqual: boolean;
}

export interface AlertmanagerSilencePayloadT {
  id?: string;
  matchers: AlertmanagerSilenceMatcherT[];
  startsAt: string;
  endsAt: string;
  createdBy: string;
  comment: string;
}

export interface APIAnnotationT {
  name: string;
  value: string;
  visible: boolean;
  isLink: boolean;
  isAction: boolean;
}

export interface APIAlertmanagerStateT {
  fingerprint: string;
  name: string;
  cluster: string;
  state: AlertStateT;
  startsAt: string;
  source: string;
  silencedBy: string[];
  inhibitedBy: string[];
}

export interface APIAlertT {
  id: string;
  annotations: APIAnnotationT[];
  labels: LabelsT;
  startsAt: string;
  state: AlertStateT;
  alertmanager: APIAlertmanagerStateT[];
  receiver: string;
}

export interface StateCountT {
  active: number;
  suppressed: number;
  unprocessed: number;
}

export interface APIAlertGroupT {
  id: string;
  receiver: string;
  labels: LabelsT;
  alerts: APIAlertT[];
  alertmanagerCount: { [key: string]: number };
  stateCount: StateCountT;
  shared: {
    annotations: APIAnnotationT[];
    labels: LabelsT;
    silences: { [cluster: string]: string[] };
  };
}

export interface APISilenceT {
  id: string;
  matchers: AlertmanagerSilenceMatcherT[];
  startsAt: string;
  endsAt: string;
  createdAt: string;
  createdBy: string;
  comment: string;
  ticketID: string;
  ticketURL: string;
}

export interface APIManagedSilenceT {
  alertCount: number;
  cluster: string;
  isExpired: boolean;
  silence: APISilenceT;
}

export interface APIGridT {
  labelName: string;
  labelValue: string;
  alertGroups: APIAlertGroupT[];
  stateCount: StateCountT;
}

export interface APIAlertmanagerUpstreamT {
  name: string;
  cluster: string;
  uri: string;
  publicURI: string;
  readonly: boolean;
  headers: { [key: string]: string };
  corsCredentials: "omit" | "same-origin" | "include";
  error: string;
  version: string;
  clusterMembers: string[];
}

export interface APIFilterT {
  text: string;
  name: string;
  matcher: string;
  value: string;
  hits: number;
  isValid: boolean;
}

export interface APIAlertsResponseUpstreamsCountersT {
  total: number;
  healthy: number;
  failed: number;
}

export type APIAlertsResponseUpstreamsClusterMapT = {
  [clusterName: string]: string[];
};

export type APIAlertsResponseSilenceMapT = {
  [clusterName: string]: { [silenceID: string]: APISilenceT };
};

export interface APIAlertsResponseAuthenticationT {
  enabled: boolean;
  username: string;
}

export interface APILabelColorT {
  brightness: number;
  background: string;
}
export type APIAlertsResponseColorsT = {
  [labelName: string]: { [labelValue: string]: APILabelColorT };
};

export interface APIAlertsResponseUpstreamsT {
  counters: APIAlertsResponseUpstreamsCountersT;
  instances: APIAlertmanagerUpstreamT[];
  clusters: APIAlertsResponseUpstreamsClusterMapT;
}

export interface APILabelCounterValueT {
  value: string;
  raw: string;
  hits: number;
  percent: number;
  offset: number;
}

export interface APILabelCounterT {
  name: string;
  values: APILabelCounterValueT[];
  hits: number;
}

export interface APISettingsT {
  staticColorLabels: string[];
  annotationsDefaultHidden: boolean;
  annotationsHidden: string[];
  annotationsVisible: string[];
  annotationsEnableHTML: boolean;
  sorting: {
    grid: {
      order: string;
      reverse: boolean;
      label: string;
    };
    valueMapping: { [labelName: string]: { [labelValue: string]: number } };
  };
  silenceForm: {
    strip: {
      labels: string[];
    };
  };
  alertAcknowledgement: {
    enabled: boolean;
    durationSeconds: number;
    author: string;
    comment: string;
  };
  historyEnabled: boolean;
}

export interface APIAlertsResponseT {
  status: string;
  error: string;
  timestamp: string;
  version: string;
  upstreams: APIAlertsResponseUpstreamsT;
  silences: APIAlertsResponseSilenceMapT;
  grids: APIGridT[];
  totalAlerts: number;
  colors: APIAlertsResponseColorsT;
  filters: APIFilterT[];
  counters: APILabelCounterT[];
  settings: APISettingsT;
  authentication: APIAlertsResponseAuthenticationT;
  receivers: string[];
}

export interface HistorySampleT {
  timestamp: string;
  value: number;
}

export interface HistoryResponseT {
  error: string;
  samples: HistorySampleT[];
}
