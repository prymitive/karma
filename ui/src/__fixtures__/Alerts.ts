import {
  APIAlertmanagerUpstreamT,
  APIAlertGroupT,
  APIAlertT,
  APISilenceT,
  APIAnnotationT,
  AlertStateT,
  LabelsT,
} from "Models/APITypes";

const MockAnnotation = (
  name: string,
  value: string,
  visible: boolean,
  isLink: boolean,
  isAction: boolean
): APIAnnotationT => ({
  name: name,
  value: value,
  visible: visible,
  isLink: isLink,
  isAction: isAction,
});

const MockAlert = (
  annotations: APIAnnotationT[],
  labels: LabelsT,
  state: AlertStateT
): APIAlertT => ({
  id: Math.random().toString(36),
  annotations: annotations,
  labels: labels,
  startsAt: "2018-08-14T17:36:40.017867056Z",
  state: state,
  alertmanager: [
    {
      fingerprint: "1234567",
      name: "default",
      cluster: "default",
      state: state,
      startsAt: "2018-08-14T17:36:40.017867056Z",
      source: "localhost/graph",
      silencedBy: [],
      inhibitedBy: [],
    },
  ],
  receiver: "by-name",
});

const MockAlertGroup = (
  rootLabels: LabelsT,
  alerts: APIAlertT[],
  sharedAnnotations: APIAnnotationT[],
  sharedLabels: LabelsT,
  sharedSilences: { [cluster: string]: string[] }
): APIAlertGroupT => ({
  receiver: "by-name",
  labels: rootLabels,
  alerts: alerts,
  id: "099c5ca6d1c92f615b13056b935d0c8dee70f18c",
  alertmanagerCount: {
    default: 1,
  },
  stateCount: {
    active: 1,
    suppressed: 0,
    unprocessed: 0,
  },
  shared: {
    annotations: sharedAnnotations,
    labels: sharedLabels,
    silences: sharedSilences,
  },
});

const MockSilence = (): APISilenceT => ({
  comment: "Mocked Silence",
  createdAt: "0001-01-01T00:00:00Z",
  createdBy: "me@example.com",
  startsAt: "2000-01-01T00:00:00Z",
  endsAt: "2000-01-01T01:00:00Z",
  id: "04d37636-2350-4878-b382-e0b50353230f",
  ticketID: "",
  ticketURL: "",
  matchers: [
    { name: "regex", value: "equal", isRegex: true, isEqual: true },
    { name: "regex", value: "notEqual", isRegex: true, isEqual: false },
    { name: "notRegex", value: "equal", isRegex: false, isEqual: true },
    { name: "notRegex", value: "notEqual", isRegex: false, isEqual: false },
  ],
});

const MockAlertmanager = (): APIAlertmanagerUpstreamT => ({
  name: "default",
  cluster: "default",
  uri: "http://localhost",
  publicURI: "http://am.example.com",
  readonly: false,
  headers: {
    Authorization: "Basic foo bar",
  },
  corsCredentials: "include",
  error: "",
  version: "0.17.0",
  clusterMembers: ["default"],
});

export {
  MockAnnotation,
  MockAlert,
  MockAlertGroup,
  MockSilence,
  MockAlertmanager,
};
