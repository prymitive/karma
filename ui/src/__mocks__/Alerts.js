const MockAnnotation = (name, value, visible, isLink) => ({
  name: name,
  value: value,
  visible: visible,
  isLink: isLink,
});

const MockAlert = (annotations, labels, state) => ({
  id: Math.random().toString(36),
  annotations: annotations,
  labels: labels,
  startsAt: "2018-08-14T17:36:40.017867056Z",
  state: state,
  alertmanager: [
    {
      name: "default",
      cluster: "default",
      state: "active",
      startsAt: "2018-08-14T17:36:40.017867056Z",
      source: "localhost/prometheus",
      silencedBy: [],
      inhibitedBy: [],
    },
  ],
  receiver: "by-name",
});

const MockAlertGroup = (
  rootLabels,
  alerts,
  sharedAnnotations,
  sharedLabels,
  sharedSilences
) => ({
  receiver: "by-name",
  labels: rootLabels,
  alerts: alerts,
  id: "099c5ca6d1c92f615b13056b935d0c8dee70f18c",
  alertmanagerCount: {
    default: 1,
  },
  stateCount: {
    active: 1,
    suppressed: 0,
    unprocessed: 0,
  },
  shared: {
    annotations: sharedAnnotations,
    labels: sharedLabels,
    silences: sharedSilences,
  },
});

const MockSilence = () => ({
  comment: "Mocked Silence",
  createdAt: "0001-01-01T00:00:00Z",
  createdBy: "me@example.com",
  startsAt: "2000-01-01T00:00:00Z",
  endsAt: "2000-01-01T01:00:00Z",
  id: "04d37636-2350-4878-b382-e0b50353230f",
  ticketID: "",
  ticketURL: "",
  matchers: [
    { name: "foo", value: "bar", isRegex: false },
    { name: "baz", value: "regex", isRegex: true },
  ],
});

const MockAlertmanager = () => ({
  name: "default",
  cluster: "default",
  uri: "http://localhost",
  publicURI: "http://am.example.com",
  readonly: false,
  headers: {
    Authorization: "Basic foo bar",
  },
  corsCredentials: "include",
  error: "",
  version: "0.17.0",
  clusterMembers: ["default"],
});

export {
  MockAnnotation,
  MockAlert,
  MockAlertGroup,
  MockSilence,
  MockAlertmanager,
};
