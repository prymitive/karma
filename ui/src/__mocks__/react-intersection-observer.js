import { useRef } from "react";

const mock = {
  value: true,
};

const useInView = () => {
  const ref = useRef(null);
  return [ref, mock.value];
};
useInView.setInView = (val) => {
  mock.value = val;
};

export { useInView };
