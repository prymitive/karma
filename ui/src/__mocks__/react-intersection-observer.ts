import { useRef } from "react";

const mock = {
  value: true,
};

const useInView = () => {
  const ref = useRef(null);
  return [ref, mock.value];
};

useInView.setInView = (val: boolean) => {
  mock.value = val;
};

export { useInView };
