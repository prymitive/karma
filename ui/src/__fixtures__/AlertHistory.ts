import { HistoryResponseT } from "Models/APITypes";

export const EmptyHistoryResponse: HistoryResponseT = {
  error: "",
  samples: Array(24).fill({ timestamp: "", value: 0 }),
};

export const FailedHistoryResponse: HistoryResponseT = {
  error: "mock error",
  samples: Array(24).fill({ timestamp: "", value: 0 }),
};

export const RainbowHistoryResponse: HistoryResponseT = {
  error: "",
  samples: Array(24)
    .fill(0)
    .map((_, i) => ({ timestamp: "", value: i % 6 })),
};
