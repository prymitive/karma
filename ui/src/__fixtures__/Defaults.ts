import { UIDefaults } from "Models/UI";

const DefaultsBase64 =
  "eyJSZWZyZXNoIjo0NTAwMDAwMDAwMCwiSGlkZUZpbHRlcnNXaGVuSWRsZSI6ZmFsc2UsIkNvbG9yVGl0bGViYXIiOmZhbHNlLCJUaGVtZSI6ImxpZ2h0IiwiQW5pbWF0aW9ucyI6dHJ1ZSwiTWluaW1hbEdyb3VwV2lkdGgiOjU1NSwiQWxlcnRzUGVyR3JvdXAiOjE1LCJDb2xsYXBzZUdyb3VwcyI6ImV4cGFuZGVkIiwiTXVsdGlHcmlkTGFiZWwiOiIiLCJNdWx0aUdyaWRTb3J0UmV2ZXJzZSI6ZmFsc2V9Cg==";
const DefaultsObject: UIDefaults = {
  Refresh: 45000000000,
  HideFiltersWhenIdle: false,
  ColorTitlebar: false,
  Theme: "light",
  Animations: true,
  MinimalGroupWidth: 555,
  AlertsPerGroup: 15,
  CollapseGroups: "expanded",
  MultiGridLabel: "",
  MultiGridSortReverse: false,
};

export { DefaultsBase64, DefaultsObject };
