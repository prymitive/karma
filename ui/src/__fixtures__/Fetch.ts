import { MockAlert, MockAlertGroup, MockSilence } from "__fixtures__/Alerts";
import {
  APIAlertsResponseT,
  APILabelCounterT,
  APIManagedSilenceT,
} from "Models/APITypes";

const EmptyAPIResponse = (): APIAlertsResponseT => ({
  status: "success",
  error: "",
  timestamp: new Date().toISOString(),
  version: "fakeVersion",
  upstreams: {
    counters: { total: 1, healthy: 1, failed: 0 },
    clusters: { default: ["default"] },
    instances: [
      {
        name: "default",
        cluster: "default",
        clusterMembers: ["default"],
        uri: "http://localhost",
        publicURI: "http://localhost",
        error: "",
        version: "0.21.0",
        readonly: false,
        corsCredentials: "include",
        headers: {},
      },
    ],
  },
  silences: { default: {} },
  grids: [],
  receivers: ["by-cluster-service", "by-name"],
  totalAlerts: 0,
  colors: {},
  filters: [
    {
      text: "label=value",
      name: "label",
      matcher: "=",
      value: "value",
      hits: 0,
      isValid: true,
    },
  ],
  counters: [] as APILabelCounterT[],
  settings: {
    sorting: {
      grid: {
        order: "startsAt",
        reverse: false,
        label: "alertname",
      },
      valueMapping: {
        cluster: {
          dev: 3,
          prod: 1,
          staging: 2,
        },
        severity: {
          critical: 1,
          info: 3,
          warning: 2,
        },
      },
    },
    silenceForm: {
      strip: {
        labels: [],
      },
    },
    alertAcknowledgement: {
      enabled: false,
      durationSeconds: 900,
      author: "karma / author missing",
      comment: "ACK! Mock comment",
    },
    staticColorLabels: ["job"],
    annotationsDefaultHidden: false,
    annotationsHidden: [],
    annotationsVisible: [],
    annotationsEnableHTML: false,
    historyEnabled: true,
  },
  authentication: {
    username: "",
    enabled: false,
  },
});

const MockAPIResponse = (): APIAlertsResponseT => {
  const response = EmptyAPIResponse();
  response.grids = [
    {
      labelName: "",
      labelValue: "",
      alertGroups: [
        MockAlertGroup(
          { alertname: "foo" },
          [MockAlert([], { instance: "foo" }, "suppressed")],
          [],
          { cluster: "dev" },
          {}
        ),
      ],
      stateCount: {
        unprocessed: 1,
        suppressed: 2,
        active: 3,
      },
    },
  ];
  response.totalAlerts = 25;
  return response;
};

const MockSilenceResponse = (
  cluster: string,
  count: number
): APIManagedSilenceT[] => {
  const silences: APIManagedSilenceT[] = [];
  for (let index = 1; index <= count; index++) {
    const silence = MockSilence();
    silence.id = `silence${index}`;
    silences.push({
      cluster: cluster,
      alertCount: 123,
      isExpired: index < 3,
      silence: silence,
    });
  }
  return silences;
};

export { EmptyAPIResponse, MockAPIResponse, MockSilenceResponse };
