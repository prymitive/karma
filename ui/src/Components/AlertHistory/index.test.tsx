import { act } from "react-dom/test-utils";

import { mount } from "enzyme";

import fetchMock from "fetch-mock";

import { useInView } from "react-intersection-observer";

import toDiffableHtml from "diffable-html";

import { MockAlertGroup, MockAlert } from "__fixtures__/Alerts";
import {
  EmptyHistoryResponse,
  RainbowHistoryResponse,
  FailedHistoryResponse,
} from "__fixtures__/AlertHistory";
import { APIAlertGroupT } from "Models/APITypes";
import { AlertHistory } from ".";

let group: APIAlertGroupT;

const MockGroup = (groupName: string) => {
  const group = MockAlertGroup(
    { alertname: "Fake Alert", groupName: groupName },
    [],
    [],
    {},
    {}
  );
  return group;
};

const MockAlerts = (alertCount: number) => {
  for (let i = 1; i <= alertCount; i++) {
    const alert = MockAlert([], { instance: `instance${i}` }, "active");
    const startsAt = new Date();
    alert.startsAt = startsAt.toISOString();
    alert.alertmanager[0].startsAt = startsAt.toISOString();
    group.alerts.push(alert);
  }
};

beforeEach(() => {
  group = MockGroup("fakeGroup");
});

describe("<AlertHistory />", () => {
  it("matches snapshot with empty response", async () => {
    fetchMock.resetHistory();
    fetchMock.mock(
      "*",
      {
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(EmptyHistoryResponse),
      },
      {
        overwriteRoutes: true,
      }
    );

    MockAlerts(3);
    const tree = mount(<AlertHistory group={group}></AlertHistory>);
    await act(async () => {
      await fetchMock.flush(true);
    });
    expect(fetchMock.calls()).toHaveLength(2);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("matches snapshot with rainbow response", async () => {
    fetchMock.resetHistory();
    fetchMock.mock(
      "*",
      {
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(RainbowHistoryResponse),
      },
      {
        overwriteRoutes: true,
      }
    );

    MockAlerts(3);
    const tree = mount(<AlertHistory group={group}></AlertHistory>);
    await act(async () => {
      await fetchMock.flush(true);
    });
    expect(fetchMock.calls()).toHaveLength(2);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("doesn't fetch when not in view", async () => {
    fetchMock.resetHistory();
    fetchMock.mock(
      "*",
      {
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(RainbowHistoryResponse),
      },
      {
        overwriteRoutes: true,
      }
    );

    (useInView as jest.MockedFunction<typeof useInView>).mockReturnValue([
      jest.fn(),
      false,
    ] as any);

    MockAlerts(3);
    const tree = mount(<AlertHistory group={group}></AlertHistory>);
    await act(async () => {
      await fetchMock.flush(true);
    });
    tree.unmount();

    expect(fetchMock.calls()).toHaveLength(0);
  });

  it("handles reponses with errors", async () => {
    fetchMock.resetHistory();
    fetchMock.mock(
      "*",
      {
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(FailedHistoryResponse),
      },
      {
        overwriteRoutes: true,
      }
    );

    MockAlerts(3);
    const tree = mount(<AlertHistory group={group}></AlertHistory>);
    await act(async () => {
      await fetchMock.flush(true);
    });
    expect(fetchMock.calls()).toHaveLength(2);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });

  it("handles fetch errors", async () => {
    fetchMock.resetHistory();
    fetchMock.mock(
      "*",
      {
        status: 500,
        body: "Error",
      },
      {
        overwriteRoutes: true,
      }
    );

    MockAlerts(3);
    const tree = mount(<AlertHistory group={group}></AlertHistory>);
    await act(async () => {
      await fetchMock.flush(true);
    });
    expect(fetchMock.calls()).toHaveLength(2);
    expect(toDiffableHtml(tree.html())).toMatchSnapshot();
  });
});
