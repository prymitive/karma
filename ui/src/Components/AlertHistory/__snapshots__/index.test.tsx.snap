// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertHistory /> handles fetch errors 1`] = `
"
<div class=\\"w-100 d-flex\\">
  <div class=\\"w-100 d-flex justify-content-between align-self-center\\">
    <div class=\\"alert-history-tooltip tooltip-trigger\\">
      <svg class=\\"alert-history\\">
        <rect rx=\\"2\\"
              ry=\\"2\\"
              class=\\"error\\"
        >
        </rect>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`<AlertHistory /> handles reponses with errors 1`] = `
"
<div class=\\"w-100 d-flex\\">
  <div class=\\"w-100 d-flex justify-content-between align-self-center\\">
    <div class=\\"alert-history-tooltip tooltip-trigger\\">
      <svg class=\\"alert-history\\">
        <rect rx=\\"2\\"
              ry=\\"2\\"
              class=\\"error\\"
        >
        </rect>
      </svg>
    </div>
  </div>
</div>
"
`;

exports[`<AlertHistory /> matches snapshot with empty response 1`] = `
"
<div class=\\"w-100 d-flex\\">
  <div class=\\"w-100 d-flex justify-content-between align-self-center\\">
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
  </div>
</div>
"
`;

exports[`<AlertHistory /> matches snapshot with rainbow response 1`] = `
"
<div class=\\"w-100 d-flex\\">
  <div class=\\"w-100 d-flex justify-content-between align-self-center\\">
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-1\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-2\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-3\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-4\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-5\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-1\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-2\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-3\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-4\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-5\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-1\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-2\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-3\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-4\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-5\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"inactive\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-1\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-2\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-3\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-4\\"
      >
      </rect>
    </svg>
    <svg class=\\"alert-history\\">
      <rect rx=\\"2\\"
            ry=\\"2\\"
            class=\\"firing firing-5\\"
      >
      </rect>
    </svg>
  </div>
</div>
"
`;
